scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/JobEmailNotification.java

```diff
@@ -136,22 +142,39 @@ public boolean doCheckAndSend(boolean withAttachment)
 
         if ((!jobStatusList.contains(eventType.toString().toLowerCase()) &&
              !jobStatusList.contains(eventType.name().toLowerCase())) &&
-            (!jobStatusList.contains(SchedulerEvent.JOB_RUNNING_TO_FINISHED_WITH_ERRORS.name().toLowerCase())) &&
-            (!jobStatusList.contains(SchedulerEvent.JOB_RUNNING_TO_FINISHED_WITH_ERRORS.toString().toLowerCase()))) {
+            (!jobStatusList.contains(eventFinishedWithErrorsName)) &&
+            (!jobStatusList.contains(eventFinishedWithErrorsMethod))) {
             return false;
         }
 
         switch (eventType) {
             case JOB_PENDING_TO_FINISHED:
             case JOB_RUNNING_TO_FINISHED:
-                if ((jobStatusList.contains(SchedulerEvent.JOB_RUNNING_TO_FINISHED_WITH_ERRORS.toString()
-                                                                                              .toLowerCase()) ||
-                     jobStatusList.contains(SchedulerEvent.JOB_RUNNING_TO_FINISHED_WITH_ERRORS.name().toLowerCase()))) {
+                // first case: check if JOB_RUNNING_TO_FINISHED is not provided along with JOB_RUNNING_TO_FINISHED_WITH_ERRORS
+                if ((jobStatusList.contains(eventFinishedWithErrorsName) ||
+                     jobStatusList.contains(eventFinishedWithErrorsMethod)) &&
+                    (!jobStatusList.contains(eventFinishedName) && !jobStatusList.contains(eventFinishedMethod))) {
+                    // check if any tasks have issues
                     if (hasTasksWithIssues()) {
                         sendEmail(withAttachment, true);
+                        //if not we do not send any notification as JOB_RUNNING_TO_FINISHED is not included
                     } else {
                         return false;
                     }
+
+                    // second case: check if JOB_RUNNING_TO_FINISHED is provided along with JOB_RUNNING_TO_FINISHED_WITH_ERRORS (e.g., the case of 'All' event)
+                } else if ((jobStatusList.contains(eventFinishedWithErrorsName) ||
+                            jobStatusList.contains(eventFinishedWithErrorsMethod)) &&
+                           (jobStatusList.contains(eventFinishedName) || jobStatusList.contains(eventFinishedMethod))) {
```

Comment:
This test 
`&& (jobStatusList.contains(eventFinishedName) || jobStatusList.contains(eventFinishedMethod))`
can be deleted since we are in the `else if` bloc.

Reply:
@mklkun that's true. I fix it. Thx))